src/core/store/common/tokensByAsset.selectors.ts

```diff
@@ -0,0 +1,86 @@
+import { createSelector } from '@reduxjs/toolkit';
+import { memoize } from 'lodash';
+
+import { toBN } from '@utils';
+import { getConfig } from '@config';
+import { TokenView, Address } from '@types';
+
+import { VaultsSelectors } from '../modules/vaults/vaults.selectors';
+import { LabsSelectors } from '../modules/labs/labs.selectors';
+import { TokensSelectors } from '../modules/tokens/tokens.selectors';
+import { AppSelectors } from '../modules/app/app.selectors';
+import { createToken } from '../modules/tokens/tokens.selectors';
+
+const { selectVaultsMap } = VaultsSelectors;
+const { selectLabsMap } = LabsSelectors;
+const { selectTokensMap, selectTokensUser } = TokensSelectors;
+const { selectServicesEnabled } = AppSelectors;
+
+export const selectDepositTokenOptionsByAsset = createSelector(
+  [selectVaultsMap, selectLabsMap, selectTokensMap, selectTokensUser, selectServicesEnabled],
+  (vaultsMap, labsMap, tokensMap, tokensUser, servicesEnabled) =>
+    memoize((assetAddress?: string): TokenView[] => {
+      if (!assetAddress) return [];
+
+      const { userTokensAddresses, userTokensMap, userTokensAllowancesMap } = tokensUser;
+      const assetData = vaultsMap[assetAddress] ? vaultsMap[assetAddress] : labsMap[assetAddress];
+      if (!assetData) return [];
+
+      const zapperDisabled = !servicesEnabled.zapper && assetData.metadata.zapInWith === 'zapperZapIn';
+      const depositTokenAddresses: Address[] = [];
+      let mainVaultToken: Address;
+      if (zapperDisabled) {
+        mainVaultToken = assetData.token;
+        depositTokenAddresses.push(mainVaultToken);
+      } else {
+        mainVaultToken = assetData.metadata.defaultDisplayToken;
+        depositTokenAddresses.push(mainVaultToken);
+        depositTokenAddresses.push(...userTokensAddresses.filter((address) => address !== mainVaultToken));
+      }
+
+      const tokens = depositTokenAddresses
+        .filter((address) => !!tokensMap[address])
+        .map((address) => {
+          const tokenData = tokensMap[address];
+          const userTokenData = userTokensMap[address];
+          const allowancesMap = userTokensAllowancesMap[address] ?? {};
+          return createToken({ tokenData, userTokenData, allowancesMap });
+        });
+      return tokens.filter((token) => toBN(token.balance).gt(0) || token.address === mainVaultToken);
+    })
+);
+
+export const selectWithdrawTokenOptionsByAsset = createSelector(
+  [selectVaultsMap, selectLabsMap, selectTokensMap, selectTokensUser, selectServicesEnabled],
+  (vaultsMap, labsMap, tokensMap, tokensUser, servicesEnabled) =>
+    memoize((assetAddress?: string): TokenView[] => {
+      if (!assetAddress) return [];
+
+      const { ZAP_OUT_TOKENS } = getConfig();
```

Comment:
This is only used in line 73, let's move it inside the `else` block

Reply:
moved here https://github.com/yearn/yearn-finance-v3/pull/674/commits/795eb535cedbb008a925adb6641da8b09fa09865